### PR TITLE
Return schema when using getTables from FlightSQL

### PR DESCRIPTION
**Pull request by jcralmeida**
When the parameter includeSchema is set, now the method getTables is returning the correct ones.

### CODE REVIEW

File: services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java [hunk 1]
```diff
@@ -253,25 +269,73 @@ public void runGetTables(FlightSql.CommandGetTables commandGetTables,
 
     final UserProtos.GetTablesResp getTablesResp = responseHandler.get();
 
-    try (VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(FlightSqlProducer.Schemas.GET_TABLES_SCHEMA,
-      allocator)) {
+    final boolean includeSchema = commandGetTables.getIncludeSchema();
```
Comment (kylep-dremio) [inline services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java:272]: Can you refactor to reduce the size of this method? Rough rule of thumb is try to keep to 100 lines per method.

File: services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java [hunk 2]
```diff
@@ -253,25 +269,73 @@ public void runGetTables(FlightSql.CommandGetTables commandGetTables,
...
+
+      final IntStream columnsRange = IntStream.range(0, getColumnsResp.getColumnsCount());
+
+      columnsRange.forEach(i -> {
+        final UserProtos.ColumnMetadata columns = getColumnsResp.getColumns(i);
+
+        final String tableName = columns.getTableName();
+        final List<Field> fields = tableToFields.computeIfAbsent(tableName, tableName_ -> new ArrayList<>());
+
+        final Field field = new Field(
```
Comment (kylep-dremio) [inline services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java:298]: Shouldn't we include the nullable information?

Comment (jcralmeida) [reply services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java:298]: I think we should for consistency. If a query is returning this values so the getTables

File: services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java (hunk 1 above)
Comment (jcralmeida) [reply services/arrow-flight/src/main/java/com/dremio/service/flight/impl/FlightWorkManager.java:272]: Nice, I've refactored

**Pull request merged**